internals/secrethub/env_source_test.go

```diff
@@ -0,0 +1,142 @@
+package secrethub
+
+import (
+	"testing"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/api/uuid"
+	"github.com/secrethub/secrethub-go/internals/assert"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
+)
+
+func TestSecretsDirEnv(t *testing.T) {
+	const dirPath = "namespace/repo"
+	testUUID1 := uuid.New()
+	testUUID2 := uuid.New()
+	testUUID3 := uuid.New()
+	testUUID4 := uuid.New()
```

Comment:
We can also define some of the secrets and directories used in multiple tests here, so that they can be re-used between tests. Not sure whether that makes things easier.

```Go
subDir := api.Dir{
  DirID:  uuid.New(),
  Name:   "foo",
}

secret := api.Secret{
  DirID:    subDir.DirID,
  SecretID: uuid.New(),
  Name:     "bar",
}
```

Reply:
Thanks for the suggestions. I improved the names of the UUIDs.

Factoring out secrets and directories used in the tests can make the code shorter, but it seems to me that it also makes the tests more difficult to read, since I have to check which secret/dir is in which directory.